ddtrace/internal/telemetry/writer.py

```diff
@@ -32,25 +32,67 @@ def _get_interval_or_default():
     return float(os.getenv("DD_TELEMETRY_HEARTBEAT_INTERVAL", default=60))
 
 
+class _TelemetryClient:
+    def __init__(self, endpoint):
+        # type: (str) -> None
+        self._agent_url = get_trace_url()
+        self._endpoint = endpoint
+        self._encoder = JSONEncoderV2()
+        self._headers = {
+            "Content-type": "application/json",
+            "DD-Telemetry-API-Version": "v1",
+        }
+
+    @property
+    def url(self):
+        return "%s/%s" % (self._agent_url, self._endpoint)
+
+    def send_event(self, request):
+        # type: (Dict) -> Optional[httplib.HTTPResponse]
+        """Sends a telemetry request to the trace agent"""
+        resp = None
+        try:
+            rb_json = self._encoder.encode(request)
+            headers = self.get_headers(request)
+            with StopWatch() as sw:
+                conn = get_connection(self._agent_url)
+                conn.request("POST", self._endpoint, rb_json, headers)
+                resp = get_connection_response(conn)
+            if resp.status < 300:
+                log.debug("sent %d in %.5fs to %s. response: %s", len(rb_json), sw.elapsed(), self.url, resp.status)
+            else:
+                log.debug("failed to send telemetry to the Datadog Agent at %s. response: %s", self.url, resp.status)
+        except Exception:
+            log.debug("failed to send telemetry to the Datadog Agent at %s.", self.url, exc_info=True)
+        finally:
+            conn.close()
+        return resp
+
+    def get_headers(self, request):
+        # type: (Dict) -> Dict
+        """Get all telemetry api v1 request headers"""
+        headers = self._headers.copy()
+        headers["DD-Telemetry-Debug-Enabled"] = request["debug"]
```

Comment:
Looks like `"DD-Telemetry-Debug-Enabled": str(self._debug).lower()` isn't transferred directly to `_TelemetryClient._headers`. Could you help me understand how this accomplishes that here?

Reply:
The "DD-Telemetry-Debug-Enabled" header is now set in `TelmeetryClient.get_headers()` instead of TelemetryWriter.__init__(). The value set should always be the same since 
[get_headers](https://github.com/DataDog/dd-trace-py/blob/1313f54e4f568f6eba1f2e048b204da437d48cb4/ddtrace/internal/telemetry/writer.py#L56) is called before every telemetry request is sent

Reply:
Yup they should be the same value. Right now they are not (now when the request body has `debug:False` the value in the header is `DD-Telemetry-Debug-Enabled:false`). This change makes sure they're consistent. This isn't a fix since the backend can accept both values. The inconsistency just didn't sit right with me